app/src/main/java/com/example/yard/activities/MainActivity.java

```diff
@@ -77,7 +85,59 @@ public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         Intent intent = new Intent(this, LoginActivity.class);
         startActivity(intent);
         return true;
+      case R.id.btnDelete:
+        deleteUser();
     }
     return super.onOptionsItemSelected(item);
   }
+
+  private void deleteUser() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this);
+    builder
+        .setTitle("DELETE ACCOUNT")
+        .setMessage("CONFIRM: Do you want to delete your account ? ")
+        .setNegativeButton(
+            "NO",
+            new DialogInterface.OnClickListener() {
+              @Override
+              public void onClick(DialogInterface dialogInterface, int i) {}
+            })
+        .setPositiveButton(
+            "YES",
+            new DialogInterface.OnClickListener() {
+              @Override
+              public void onClick(DialogInterface dialogInterface, int i) {
+                ParseUser user = ParseUser.getCurrentUser();
+                ParseQuery<Post> query = new ParseQuery<>("Post");
+                query.whereEqualTo(Post.KEY_USER, ParseUser.getCurrentUser());
+                query.findInBackground(
+                    new FindCallback<Post>() {
+                      @Override
+                      public void done(List<Post> objects, ParseException e) {
+                        for (Post object : objects) {
+                          object.deleteInBackground(
```

Comment:
Is there an API that can do bulk deleting? This is not very efficient though. 
Is it possible to customize an API that delete all posts assigned to a user id?